### PR TITLE
Feature/VTA-336 - Handle response from submitting a visit

### PR DESCRIPTION
**Pull request by FBarrassDVSA**
## Description
Moved VTA-336 work from Ionic 3 to Ionic 6 with a bit or refactoring

Related issue: [VTA-336](https://dvsa.atlassian.net/browse/VTA-336)

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code
- [x] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: src/app/pages/visit/visit-timeline/visit-timeline.spec.ts [hunk 1]
```diff
@@ -65,8 +67,9 @@ describe('Component: VisitTimelinePage', () => {
   let logProviderSpy: any;
   let analyticsService: AnalyticsService;
   let analyticsServiceSpy: any;
-  const waitActivity = ActivityDataMock.WaitActivityData;
+  let router;
```
Comment (tomcrawleyy) [inline src/app/pages/visit/visit-timeline/visit-timeline.spec.ts:70]: should be type strict

**Review by tomevs88 — APPROVED** [review]

**Review by tomcrawleyy — APPROVED** [review]

**Comment by FBarrassDVSA** [comment]
No internet:
image

**Comment by FBarrassDVSA** [comment]
504:
image

**Comment by FBarrassDVSA** [comment]
400:
image

**Comment by FBarrassDVSA** [comment]
Call IT:
image

**Pull request merged**